### DISCUSSION TITLE
Project Promotion Strategy

### DISCUSSION THREAD

**Discussion by panno8M** [godot-nim/.github#7] [Ideas]
@insomniacUNDERSCORElemon 
This is in response to https://github.com/godot-nim/.github/issues/1#issuecomment-2267260756

I have been thinking about this too and would love to hear ideas.

1. The dispersion of interest due to scattered repositories will be addressed by making gdextwiz the main repository.
1. Instead of creating gdextgen locally, create it remotely in advance using API data from godot official. (This is the work required to install the entire project with a single `nimble install`.)
1. Specify godot-nim/[gdextcore, gdextgen, gdext] as the requires of gdextwiz.nimble.
1. panno8M/godot-nim will be public-archived with an emphasis on directions to gdoot-nim.org.

We believe that this is the best way to prevent dispersion of interest and provide a good lead-in. What do you think?

**Comment by panno8M** [comment]
### Reasons for making panno8M/godot-nim a public archive

ertainly the number of stars now is overwhelming compared to that of godot-nim.org, but we don't want it to be that way forever.

Perhaps transferring ownership from panno8M to godot-nim.org would be a better outcome, but I have an evil thought that since I have received so many stars, I want to keep them as a personal achievement.

**Comment by insomniacUNDERSCORElemon** [comment]
Some scattered thoughts:

I was thinking you keep going with this seeing where it goes, perhaps you can mirror at the old repo at a later date and it may make a bigger difference for the watchers who--for whatever reason--have not checked out the new setup.

I am also not sure a mirror won't just add to the complexity. Unless it is possible to have some sort of repo-shifting of issues or a triage reposting bot or something like that. Or if you turn off issues/PR in the old repo.

If the old repo keeps gaining stars and not this, a mirror may be unavoidable (I am not sure why that would happen, though perhaps it's a numbers game and may not result in real activity). Perhaps you could fork and archive it if you do mirror there.

EDIT: I've hinted at it before, I'd post on the fediverse about the project if my instance weren't broken and if I didn't lack luck with Nim-lang there (not for a lack of trying, search for it and you'll find many of my comments). I looked into but never decided on another instance, especially if they don't have Oauth. The other things I tried to post about (rarely) were niche, too.

**Comment by insomniacUNDERSCORElemon** [comment]
Now that you have changed the name, one option would be to tell relevant content creators about gdext-nim. The Godot, Nim, and general programming circles and even more so for creators where these circles overlap (this might even happen on its own).

However... starting small might actually be better, allowing to adjust to contributors/users and using said community to improve first impressions.

That said, one of the creators that I watch has 250K subscribers (and they have 2 videos on Nim: 29K and 42K views), though I don't know how well any of those numbers actually resulted in users/contributors. Those numbers seem big to me (at least for where the project is now), and it might fare better with more demos and/or easier usage (/community development in general). In the video they made that included the 3.X bindings for Godot, they talked more about Enu (which has screenshots, code snippets, videos etc).

Another creator (~20K subs) made a video on the 3.X Nim bindings (with tutorial) at 3K views, I'd guess that may be a better start though I commented there before when the video was launched but there was an issue with YT deleting comments (that was ~2 years ago, so this may not be the case now).
